src/o365/spo/commands/apppage/apppage-add.spec.ts

```diff
@@ -0,0 +1,275 @@
+    assert.equal(command.name.startsWith(commands.APPPAGE_ADD), true);
+  });
+
+  it('has a description', () => {
+    assert.notEqual(command.description, null);
+  });
+  it('calls telemetry', (done) => {
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: {} }, () => {
+      try {
+        assert(trackEvent.called);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('logs correct telemetry event', (done) => {
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: {} }, () => {
+      try {
+        assert.equal(telemetry.name, commands.APPPAGE_ADD);
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('aborts when not logged in to a SharePoint site', (done) => {
+    auth.site = new Site();
+    auth.site.connected = false;
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: { debug: true } }, (err?: any) => {
+      try {
+        assert.equal(JSON.stringify(err), JSON.stringify(new CommandError('Log in to a SharePoint Online site first')));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  it('creates a single-part app page', (done) => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if (opts.url.indexOf(`_api/sitepages/Pages/CreateFullPageApp`) > -1 &&
+     opts.body.webPartDataAsJson ===
+      "{}" && !opts.body.addToQuickLaunch)
+      {
+        return Promise.resolve("Done");
+      }
+      return Promise.reject('Invalid request');
+    });
+
+    auth.site = new Site();
+    auth.site.connected = true;
+    auth.site.url = 'https://contoso.sharepoint.com';
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: { debug: true, title: 'test-single', webUrl: 'https://contoso.sharepoint.com/',webPartData:JSON.stringify({})} }, () => {
+      try {
+       assert(cmdInstanceLogSpy.calledWith("Done"));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+  it('creates a single-part app pageshowing on quicklaunch', (done) => {
+    sinon.stub(request, 'post').callsFake((opts) => {
+      if (opts.url.indexOf(`_api/sitepages/Pages/CreateFullPageApp`) > -1 &&
+     opts.body.webPartDataAsJson ===
+      "{}" && opts.body.addToQuickLaunch)
+      {
+        return Promise.resolve("Done");
+      }
+      return Promise.reject('Invalid request');
+    });
+
+    auth.site = new Site();
+    auth.site.connected = true;
+    auth.site.url = 'https://contoso.sharepoint.com';
+    cmdInstance.action = command.action();
+    cmdInstance.action({ options: { debug: true, addToQuickLaunch:true, title: 'test-single', webUrl: 'https://contoso.sharepoint.com/',webPartData:JSON.stringify({})} }, () => {
+      try {
+       assert(cmdInstanceLogSpy.calledWith("Done"));
+        done();
+      }
+      catch (e) {
+        done(e);
+      }
+    });
+  });
+
+  
```

Comment:
Could you please add test in case of a request failure. It is optional, I can add it when I am merging. Thanks!